cmd/kube-vip.go

```diff
@@ -200,48 +203,88 @@ var kubeVipManager = &cobra.Command{
 	Short: "Start the kube-vip manager",
 	Run: func(cmd *cobra.Command, args []string) {
 
+		// Set the logging level for all subsequent functions
+		log.SetLevel(log.Level(logLevel))
+
+		go servePrometheusHTTPServer(cmd.Context(), PrometheusHTTPServerConfig{
+			Addr: initConfig.PrometheusHTTPServer,
+		})
+
 		log.Infof("Starting kube-vip.io [%s]", Release.Version)
 		// parse environment variables, these will overwrite anything loaded or flags
 		err := kubevip.ParseEnvironment(&initConfig)
 		if err != nil {
 			log.Fatalln(err)
 		}
 
+		// Determine the kube-vip mode
+		var mode string
+		if initConfig.EnableARP {
+			mode = "ARP"
+		}
+
+		if initConfig.EnableBGP {
+			mode = "BGP"
+		}
+
+		if initConfig.EnableWireguard {
+			mode = "Wireguard"
+		}
 		// Provide configuration to output/logging
-		log.Infof("namespace [%s], Mode(s): Control Plane:[%t], Services:[%t]", initConfig.Namespace, initConfig.EnableControlPane, initConfig.EnableServices)
+		log.Infof("namespace [%s], Mode: [%s], Features(s): Control Plane:[%t], Services:[%t]", initConfig.Namespace, mode, initConfig.EnableControlPane, initConfig.EnableServices)
 
 		// End if nothing is enabled
 		if !initConfig.EnableServices && !initConfig.EnableControlPane {
 			log.Fatalln("no modes are enabled")
 		}
 
-		if err := initConfig.CheckInterface(); err != nil {
-			log.Fatalln(err)
-		}
-
-		// Set the logging level for all subsequent functions
-		log.SetLevel(log.Level(logLevel))
-
-		if initConfig.Interface == "" {
-			log.Infof("No interface is specified for VIP in config, auto-detecting default Interface")
-			defaultIF, err := vip.GetDefaultGatewayInterface()
+		// If we're using wireguard then all traffic goes through the wg0 interface
+		if initConfig.EnableWireguard {
+			log.Infof("configuring Wireguard networking")
+			l, err := netlink.LinkByName("wg0")
```

Comment:
Have you considered consuming the `--interface` flag and use wg0 as a default in case unset?

Reply:
Done.